src/js/components/productSupplier/modals/PreferenceTypeModal.jsx

```diff
@@ -0,0 +1,135 @@
+import React from 'react';
+
+import PropTypes from 'prop-types';
+import { RiCloseFill } from 'react-icons/all';
+import Modal from 'react-modal';
+
+import Button from 'components/form-elements/Button';
+import { PRODUCT_SUPPLIER_URL } from 'consts/applicationUrls';
+import Translate from 'utils/Translate';
+
+const PreferenceTypeModal = ({
+  isOpen,
+  closeModal,
+  modalData,
+  productSupplierId,
+}) => {
+  const mapPreferenceTypes = (preferenceTypes) => {
+    const defaultPreferenceTypePlaceholder = (
+      <p className="default-preference-type">
+        <span className="preference-type-location">Default: </span>
+      </p>
+    );
+
+    return preferenceTypes?.reduce((acc, preferenceType) => {
+      const {
+        preferenceType: { name },
+        destinationParty,
+      } = preferenceType;
+
+      if (!destinationParty) {
+        return {
+          ...acc,
+          default: (
+            <p className="default-preference-type">
+              <span className="preference-type-location">Default: </span>
+              <span>{name}</span>
+            </p>
+          ),
```

Comment:
returning HTML like that feels weird to me - if it would not be difficult to change, how about just storing the `name` and `destinationParty.name` in the list and then mapping it in the JSX output (85th line)?
If it is more complicated than I think, then I guess you can skip it.